ghc-lib-gen/src/Main.hs

```diff
@@ -187,6 +187,7 @@ parserModules =
   , "GhcMonad"
   , "GhcPrelude"
   , "HaddockUtils"
+  , "HeaderInfo"
```

Comment:
Is this change unrelated but just going along at the same time? (Fine with that, just checking its not a mistake that went along for the ride)

Reply:
`HeaderInfo` was in `ghc-lib`. I moved it to `ghc-lib-parser` so that `hlint` is not required to depend on `ghc-lib` but can remain satisfied with `ghc-lib-parser`.